common/browser.go

```diff
@@ -255,8 +255,17 @@ func (b *Browser) onAttachedToTarget(ev *target.EventAttachedToTarget) {
 					ev.SessionID, evti.TargetID, b.ctx.Err())
 				return // ignore
 			default:
-				k6ext.Panic(b.ctx, "creating a new background page: %w", err)
 			}
+			// If the session is closed, then ignore the error and return early.
+			// This can happen if the page is closed before the attachedToTarget
+			// event is handled. Another VU or instance can close the page.
```

Comment:
Could you clarify how "Another VU or instance can close the page"? I'm assuming it's only ones that are related to the same `session`?

Reply:
Does the following clarify your confusion? Let me know if it doesn't help :)

I explain this in the PR description.

```
                                          browser pages
instance1 ---> newPage                    page1
  attaches to: page1
instance2
  attaches to: page1
instance2 ---> newPage                    page1, page2
  attaches to: page2
instance1
  attaches to: page2
instance1 --> page1.Close()               page2
instance1 <-- detachedFromTarget(page1)
  closes page1 session.

--- THIS IS THE RACY PART (happens concurrently) ---
instance2 <-- detachedFromTarget(page1)
  closes page1 session.
instance2 ---> attachToTarget(page1)
  panic: session does not exist.

instance2 panics while trying to send CDP messages to a closed web socket connection since it detaches from it and tries to attach to it concurrently.
```

Reply:
I'm also writing a test to validate this behavior.